libkbfs/ongoing_work_limiter.go

```diff
@@ -0,0 +1,40 @@
+package libkbfs
+
+import "sync"
+
+// OngoingWorkLimiter limits maximum number of routines that can work on a same
+// type of thing at the same time. For example, it can be used for limiting
+// number of ongoing rekeys.
+type OngoingWorkLimiter struct {
+	cond     sync.Cond
+	capacity int64
+}
+
+// NewOngoingWorkLimiter creates a new *OngoingWorkLimiter with capacity of
+// maxNumOngoingWorks.
+func NewOngoingWorkLimiter(maxNumOngoingWorks int) *OngoingWorkLimiter {
+	return &OngoingWorkLimiter{
+		capacity: int64(maxNumOngoingWorks),
```

Comment:
Why not just require an `int64` to be passed in?  (And why an `int64` instead of an `int` anyway?  Do you think a 32-bit machine will fire up more than 2^32 workers?)